File: javascript/0050. Pow(x, n).js
const myPow = (x, n) => {
  if (n === 0) return 1;
  if (n === 1) return x;

  if (n < 0) {
    n = -n;
    x = 1 / x;
  }

  return n % 2 === 0
    ? myPow(x * x, Math.floor(n / 2))
    : x * myPow(x * x, Math.floor(n / 2));
};

console.log(myPow(2, 10));

File: python/0050. Pow(x, n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if x == 0:
            return x

        if n < 0:
            return self.myPow(1 / x, -n)

        result = 1
        curr_pow = x

        while n:
            if n & 1:
                result *= (n & 1) * curr_pow
            curr_pow *= curr_pow
            n >>= 1

        return result
